src/controllers/appController.ts

```diff
@@ -1,100 +1,57 @@
-      domain: process.env.DOMAIN || 'localhost:3000',
-      nonce: params.message.nonce,
-      signature: params.signature,
-      time: params.message.issuedAt,
-    };
-
-    // Verify using the static verify method of SiweMessage
-    const { success } = await params.message.verify(fields);
-
-    if (!success) {
-      throw new Error('Signature verification failed');
-    }
-
-    // Check expiration if present
-    if (params.message.expirationTime) {
-      const expirationTime = new Date(params.message.expirationTime);
-      if (expirationTime < new Date()) {
-        throw new Error('Message has expired');
-      }
-    }
-
-    return { address: params.message.address };
-  } catch (error) {
-    throw new Error('Invalid signature or message format');
-  }
-}
+import { verifySIWEMessage } from '../siwe/siweVerification';
 
 // Define Zod schemas for input validation
 const registerAppSchema = z.object({
-  appDescription: z.string(),
-  appName: z.string(),
-  email: z.string().email(),
-  signedMessage: z.object({
-    message: z.object({
-      address: z.string(),
-      chainId: z.number(),
-      domain: z.string(),
-      expirationTime: z.string().optional(),
-      issuedAt: z.string(),
-      nonce: z.string(),
-      statement: z.string(),
-      uri: z.string(),
-      version: z.string(),
-    }),
-    signature: z.string(),
-  }),
+  contactEmail: z.string().email(),
+  description: z.string(),
+  name: z.string(),
+  signedMessage: z.string(),
 });
 
 export const registerApp = async (req: Request, res: Response) => {
   try {
-    const { appDescription, appName, email, signedMessage } = registerAppSchema.parse(req.body);
+    const { contactEmail, description, name, signedMessage } = registerAppSchema.parse(req.body);
 
-    const siweMessage = new SiweMessage(signedMessage.message);
-    const { address: managementWallet } = await verifySIWEMessage({
-      message: siweMessage,
-      signature: signedMessage.signature
-    });
+    let managementWallet: string;
+
+    try {
+      managementWallet = await verifySIWEMessage(signedMessage);
+    } catch (error) {
+      consola.error('ERROR: ', (error as Error).message);
+      res.status(401).json({ message: (error as Error).message, success: false });
+      return;
+    }
 
     // Check if the management wallet is already registered
     const existingApp = await App.findOne({ managementWallet });
     if (existingApp) {
       res.status(200).json({ message: 'Management wallet already registered', success: true });
       return;
     }
-    const appId = generateUniqueId();
     const newApp = new App({
-      appId,
+      contactEmail,
+      description,
       managementWallet,
-      contactEmail: email,
-      description: appDescription,
-      name: appName,
+      name,
     });
 
     await newApp.save();
 
-    res.json({ data: { appId, appName }, success: true });
+    res.json({ data: { app: newApp.toObject() }, success: true });
```

Comment:
```suggestion
    res.json({ data: { app: newApp }, success: true });
```
Express automatically calls JSON.stringify() on objects passed to res.json(), thus triggering the document’s toJSON() method which is better suited for serialization than `toObject`

Reply:
Same for repetitions of this pattern